edx/edx-ora2
# [BD-5] [TNL-7282] [BB-2621] Support flexible peer grading

Add support for Flexible Peer Grading Averaging. When enabled, a submission that is waiting more than 7 days, and already got at least {floor-30% of required peer-reviews configuration value} assessments, then that submission will be graded.

**JIRA tickets**: 
- https://openedx.atlassian.net/browse/TNL-7282
- https://tasks.opencraft.com/browse/BB-2621

~~**Discussions**:~~

**Dependencies**: None

**Screenshots**: 
![localhost_18010_container_block-v1_edX+DemoX+Demo_Course+type@vertical+block@51aeb9ff0cbb4aa6a4e759fee133325a_action=new](https://user-images.githubusercontent.com/1010244/96540205-e9c4b100-12be-11eb-8b69-4c0d19fdc147.png)


~~**Sandbox URL**:~~ 

**Merge deadline**: None

**Testing instructions**:

1. Setup this PR in your local devstack.
2. Create 5 different users.
3. Create an Open Assessment question with flexible peer grading is true, Must Grade to 4 & Graded By to 4.
4. Make user1 submission.
5. Make user2 submission and a peer assessment (User 1 submission)
6. Manually change submitted_at to a date older than 7 days
7. If you visit the user1 ORA page, and complete peer assessment (4 assessments), it should show grade even if only 1 assessment received. (because floor(30% of 4) = 1)
8. For any other case, such ass flexible peer grading is false or submitted_at is not older than 7 days it should not show grade.


**Author notes and concerns**: N/A


**Reviewers**

~~**Settings**~~
